neurolib/models/model.py

```diff
@@ -179,7 +177,23 @@ def run(
                 logging.warn(f"{self.name}: BOLD model not initialized, not simulating BOLD. Use `run(bold=True)`")
                 bold = False
             self.integrateChunkwise(chunksize=chunksize, bold=bold, append_outputs=append)
-            return
+
+        # check if there was a problem with the simulated data
+        self.checkOutputs()
+
+    def checkOutputs(self):
+        # check nans in output
+        if np.isnan(np.sum(self.output)):
+            logging.error("nan in model output!")
+        else:
+            EXPLOSION_THRESHOLD = 1e20
+            if np.sum(self.output > EXPLOSION_THRESHOLD) > 0:
+                logging.error("nan in model output!")
+
+        # check nans in BOLD
+        if "BOLD" in self.outputs:
+            if np.isnan(np.sum(self.outputs.BOLD.BOLD)):
+                logging.error("nan in BOLD output!")
```

Comment:
the logging errors just write an error, is that ok? don't you want to throw an actual error or exception?

Reply:
Yep, I'm aware of this. I don't want to raise an exception though, because the original signal (not bold) could still be fine and I don't want to upset `pypet` during explorations... :/

Reply:
that's true, good catch... maybe I would then set full BOLD time series to NaNs... just in case

Reply:
Sometimes, the bold signal "crashes" mid simulation when for example the system undergoes a random bifurcation (jump from down to up). Often, _some_ notes will go `nan` and others still work because of their connection strengths. To be able to debug these cases, I'll let it just throw this message for now.... 

I wanna implement a second BOLD mechanism using the lowpass-filtered envelope of the signal some day, hope that it won't have these issues anymore.